src/components/DiagramFrame/MetadataPanel/MetadataPanel.tsx

```diff
@@ -0,0 +1,196 @@
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+
+import React from 'react';
+import {
+  Badge,
+  ButtonTransparent,
+  Code,
+  Flex,
+  FlexItem,
+  Heading,
+  SpaceVertical,
+  Paragraph,
+  Icon,
+  Space,
+  Tabs,
+  Tab,
+  TabList,
+  TabPanel,
+  TabPanels,
+  theme
+} from "@looker/components"
+import { ILookmlModel, ILookmlModelExplore } from "@looker/sdk"
+import { ExternalLink } from "../../ExternalLink"
+import JoinIcon from "./JoinIcon"
+import MetadataPanelTable from "./MetadataPanelTable"
+import {MetadataInfoPanel} from "./MetadataInfoPanel"
+import {MetadataFooter} from "./MetadataFooter"
+import {LookmlCodeBlock} from "./LookmlCodeBlock"
+import { METADATA_PANEL_PIXEL } from "../../../utils/constants"
+import { LookmlObjectMetadata, SelectionInfoPacket } from "../../interfaces"
+import { ILookmlModelExploreField, ILookmlModelExploreJoins } from '@looker/sdk/lib/sdk/3.1/models';
+import { getFields } from '../../../utils/diagrammer'
+import { exploreFieldURL } from '../../../utils/urls'
+import { useExplore } from '../../../utils/fetchers'
+import {getJoinMetadata, getFieldMetadata, getViewMetadata, getExploreMetadata} from "./utils"
+
+export const PillText:React.FC = ({children}) => {
+  return <Code color="text3" fontSize="xsmall" lineHeight="medium">{children}</Code>
+}
+
+export const MetadataPanel: React.FC<{
+  explore: ILookmlModelExplore,
+  selectionInfo: SelectionInfoPacket,
+  model: ILookmlModel,
+}> = ({
+  explore,
+  selectionInfo,
+  model
+}) => {
+  let metadata: LookmlObjectMetadata
+  let field: ILookmlModelExploreField
+  // @ts-ignore
+  let exploreLookmlLink = explore.lookml_link
+  if (selectionInfo.lookmlElement === "explore") {
+    metadata = getExploreMetadata(explore, exploreLookmlLink)
+  } else if (selectionInfo.lookmlElement === "join") {
+    let joinObj = explore.joins.filter((join: ILookmlModelExploreJoins) => {
+      return join.name === selectionInfo.name
+    })[0]
+    metadata = getJoinMetadata(joinObj, exploreLookmlLink)
+  } else if (selectionInfo.lookmlElement === "dimension" || selectionInfo.lookmlElement === "measure") {
+    let fields = getFields(explore.fields).filter((field: any) => {
```

Comment:
I think the "any" here is a LookmlModelExploreField, do you have access to looker's types in here?

Reply:
Yes, but for some reason `dimension_group` is not defined on the type definition for ILookmlModelExploreField ... I should file as a bug :slightly_smiling_face: